doc/frequency_evaluation_framework.md

```diff
@@ -0,0 +1,373 @@
+   </td>
+   <td>Create a <code>FrequencyReportGenerator</code> class for generating reports from frequency comparisons, similar to existing <code>ReportGenerator</code>.
+   </td>
+  </tr>
+</table>
+
+#### Simulator
+
+The
+[Simulator](https://github.com/world-federation-of-advertisers/cardinality_estimation_evaluation_framework/blob/master/src/simulations/simulator.py)
+currently produces a DataFrame containing four columns: `num_sets`,
+`estimated_cardinality, true_cardinality, relative_error`. The simulator will be
+modified to produce a DataFrame with 3_f_+2 columns, where _f_ is the number of
+frequencies being evaluated. The columns `estimated_cardinality,
+true_cardinality`and`relative_error`will be respectively renamed to
+<code>estimated_cardinality_<em>i</em>, true_cardinality_<em>i</em>,</code>
+<code>relative_error_<em>i</em></code>, where <em>i</em> is a frequency. The
+value of <code>estimated_<em>i</em></code> will be the estimated number of
+elements having a frequency of <em>i or higher</em>. Thus,
+<code>estimated_cardinality_1</code> will have the same meaning as the current
+<code>estimated_cardinality</code>. Similarly, <code>true_cardinality_1</code>
+will be the true number of elements having a frequency of <em>i</em> or more.
+
+In addition, a new column will be added named `shuffle_distance`.  Code will be written to compute the shuffle distance and record it in the DataFrame produced by the Simulator.
+
+
+#### SketchEstimator
+
+The
+<code>[EstimatorBase](https://github.com/world-federation-of-advertisers/cardinality_estimation_evaluation_framework/blob/master/src/estimators/base.py)</code>
+class will be redefined so that it reports a list of values. If <code>h[]</code>
+is the list of values returned by <code>EstimatorBase</code>, then
+<code>h[i]</code> is defined as the number of id's that occurred with a
+frequency of <em>i+1 or higher</em>. Thus, <code>h[0]</code> is the cardinality
+of the set, so this redefinition is almost backward compatible with the current
+definition.
+
+There are at least two different types of frequency estimation sketches in
+consideration. One type of sketch supports unions and also supports reporting of
+frequency histograms for unions. For this type of sketch, frequency estimation
+is easy.
+
+Another type of data structure under consideration consists of one cardinality
+estimation sketch per frequency. Let's call this a stratified sketch.
+Stratification is of interest because it provides a generic mechanism by which
+any cardinality estimator can be converted into a frequency estimator. To
+calculate the number of elements of frequency _f_ in the union of 2 stratified
+sketches requires considering every pair $$i, j$$ such that $$i+j=f$$. In other
+words, if if $$X_{(i)}$$ represents the elements in $$X$$ having frequency
+$$i$$, then we have $$(A \cup B)_{f}=(A_{(f)} \cap B_{(0)})\cup(A_{(f-1)} \cap
+B_{(1)}) \cup (A_{(0)} \cap B_{(f)}).$$
```

Comment:
Same as above for the math formulas here.

Reply:
This equation seems to be beyond the abilities of the math rendering widget you pointed me to, so I decided just to create an image and inline it.  Hope that's ok.